app/src/main/java/org/kiwix/kiwixmobile/localFileTransfer/WifiDirectManager.kt

```diff
@@ -57,13 +57,13 @@ import javax.inject.Inject
 /**
  * Manager for the Wifi-P2p API, used in the local file transfer module
  */
-@SuppressWarnings("MissingPermission")
+@SuppressWarnings("MissingPermission", "ProtectedMemberInFinalClass")
 class WifiDirectManager @Inject constructor(
   private val activity: Activity,
   private val sharedPreferenceUtil: SharedPreferenceUtil,
   private val alertDialogShower: AlertDialogShower
 ) : ChannelListener, PeerListListener, ConnectionInfoListener, P2pEventListener {
-  private val callbacks = activity as Callbacks
+  protected var callbacks: Callbacks? = null
```

Comment:
when do these callbacks get set? I might just have code blindness from these damn codecov annotations

Reply:
>  I might just have code blindness from these damn codecov annotations
This makes me go rofl

Reply:
Hmm, I remember clearly it being set somewhere but now I can't find it either..